prow/cmd/deck/pr_history.go

```diff
@@ -0,0 +1,247 @@
+	jobHistLinkTemplate  = "/job-history/%s/pr-logs/directory/%s"
+	// expectedPRKey indicates the key format we expect given the GCS layout specified here:
+	// https://github.com/kubernetes/test-infra/tree/master/gubernator#gcs-bucket-layout
+	expectedPRKey = "<bucket-name>/pr-logs/pull/<org>_<repo>/<pr-number>"
+	expectedPull  = "branch:hash,pullNumber:hash"
+)
+
+var prRe = regexp.MustCompile(`^([-\w]+)/pr\-logs/pull/([-/:\.\w]+_)?([-\w]+/)?([0-9]+)/?$`)
+var pullCommitRe = regexp.MustCompile(`^[-\w]+:\w{40},[0-9]+:(\w{40})$`)
+
+type prHistoryTemplate struct {
+	Link    string
+	Name    string
+	Jobs    []prJobData
+	Commits []commitData
+}
+
+type prJobData struct {
+	Name   string
+	Link   string
+	Builds []buildData
+}
+
+type commitData struct {
+	Hash       string
+	HashPrefix string // used only for display purposes, so don't worry about uniqueness
+	Link       string
+	MaxWidth   int
+	latest     time.Time // time stamp of the job most recently started
+}
+
+type latestCommit []commitData
+
+func (a latestCommit) Len() int           { return len(a) }
+func (a latestCommit) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a latestCommit) Less(i, j int) bool { return a[i].latest.Before(a[j].latest) }
+
+type byStarted []buildData
+
+func (a byStarted) Len() int           { return len(a) }
+func (a byStarted) Swap(i, j int)      { a[i], a[j] = a[j], a[i] }
+func (a byStarted) Less(i, j int) bool { return a[i].Started.Before(a[j].Started) }
+
+func githubPRLink(org, repo, prNumber string) string {
+	return fmt.Sprintf(githubPRTemplate, org, repo, prNumber)
+}
+
+func githubCommitLink(org, repo, commitHash string) string {
+	return fmt.Sprintf(githubCommitTemplate, org, repo, commitHash)
+}
+
+func jobHistLink(bktName, jobName string) string {
+	return fmt.Sprintf(jobHistLinkTemplate, bktName, jobName)
+}
+
+// gets the pull commit hash from metadata
+func getPullCommitHash(pull string) (string, error) {
+	match := pullCommitRe.FindStringSubmatch(pull)
+	if len(match) != 2 {
+		return "", fmt.Errorf("unable to parse pull %q (expected %q)", pull, expectedPull)
+	}
+	return match[1], nil
+}
+
+func parsePRHistKey(key, defaultOrg, defaultRepo string) (bktName, org, repo, prNumber string, err error) {
+	match := prRe.FindStringSubmatch(key)
+	if len(match) != 5 {
+		err = fmt.Errorf("unrecognized directory layout: %s (expected format %s)", key, expectedPRKey)
+		return
+	}
+	bktName = match[1]
+	org = match[2]
+	if org == "" {
+		org = defaultOrg
+	} else {
+		org = strings.TrimSuffix(org, "_")
+	}
+	repo = match[3]
+	if repo == "" {
+		repo = defaultRepo
+	} else {
+		repo = strings.TrimSuffix(repo, "/")
+	}
+	prNumber = match[4]
+	return
+}
+
+func getPRHistory(url *url.URL, config *config.Config, gcsClient *storage.Client) (prHistoryTemplate, error) {
+	start := time.Now()
+	tmpl := prHistoryTemplate{}
+
+	key := strings.TrimPrefix(url.Path, "/pr-history/")
+	defaultOrg := config.Plank.DefaultDecorationConfig.GCSConfiguration.DefaultOrg
```

Comment:
You should honor the GCSConfiguration strategy for parsing the org and repo correctly

Reply:
My implementation of `parsePRHistKey` covers all three strategies, so I'm not sure what the advantage would be